Zombie/src/main/java/dk/gruppeseks/bodtrd/zombie/ZombieProcessor.java

```diff
@@ -133,4 +139,49 @@ private void attackPlayer(World world, Entity zombie, Entity player)
             zWep.setAttackCooldown(zWep.getAttackSpeed());
         }
     }
+
+    private void rayCast(Entity zombie)
+    {
+        ArrayList<Float> shape = new ArrayList<>();
+        double rotSteps = 360 / RAY_COUNT;
+
+        for (int i = 0; i < RAY_COUNT; i++)
+        {
+            double rotAmount = rotSteps * i;
+            Vector2 ray = new Vector2(0, 1);
+            ray.setMagnitude(VISION_DISTANCE);
+            ray.rotateDegrees(rotAmount);
+            Position collision = getFirstCollision(ray, zombie.get(Position.class));
+            shape.add((float)collision.getX());
+            shape.add((float)collision.getY());
+        }
+        float[] floatShape = new float[shape.size()];
+        for (int i = 0; i < shape.size(); i++)
+        {
+            floatShape[i] = shape.get(i);
+        }
+
+        zombie.get(View.class).setShape(floatShape);
+    }
+
+    private Position getFirstCollision(Vector2 ray, Position start)
+    {
+        boolean[][] grid = _world.getMap().getGrid();
+        int cellSize = _world.getMap().getGridCellSize();
+
+        for (int i = 0; i < ray.getMagnitude(); i += 2)
```

Comment:
Check collision only every x (2) pixels for performance. Increase x for better performance but greater risk of missing a collision